src/commands/periscope/helpers/periscopehelper.ts

```diff
@@ -214,6 +216,105 @@ export async function generateDownloadableLinks(
     }
 }
 
+export async function downloadableLinks(
+    periscopeStorage: PeriscopeStorage,
+    output: string,
+    nodeName: string
+): Promise<any> {
+
+    try {
+        const storageAccount = periscopeStorage.storageName;
+        const storageKey = periscopeStorage.storageKey;
+
+        // Get DNS Core hostname which Periscope use it as name of the container.
+        const matches = output.match(/(https?:\/\/[^\s]+)/g);
+        let containerName;
+        if (matches![0].indexOf('://') !== -1) {
+            containerName = matches![0].replace('https://', '').split('.')[0];
+        }
+
+        // Use SharedKeyCredential with storage account and account key
+        const sharedKeyCredential = new StorageSharedKeyCredential(storageAccount, storageKey);
+        const blobServiceClient = new BlobServiceClient(
+            `https://${storageAccount}.blob.core.windows.net`,
+            sharedKeyCredential
+        );
+
+        // Hide this all under single pupose funct which lik elike : getZipDir or get me dirname
+        const containerClient = blobServiceClient.getContainerClient(containerName);
+
+        // List all current blob.
+        const listCurrentUploadedFolders = [];
+        for await (const item of containerClient.listBlobsByHierarchy("/")) {
+            if (item.kind === "prefix") {
+                listCurrentUploadedFolders.push(item.name);
+            }
+        }
```

Comment:
You may be able to replace this loop with something like `const currentFolders = (await containerClient.list(...)).filter(item => item.kind === prefix).map(item => item.name)` but I am not sure if that enumeration is async - in that case filter and map may not be available.